client/src/components/ContentNavigator/convert.ts

```diff
@@ -0,0 +1,201 @@
+    const idNode = `id-${Date.now()}-${Math.floor(Math.random() * 10000)}`;
+    arrayIdNode.push(idNode);
+    const idNote = `id-${Date.now()}-${Math.floor(Math.random() * 10000)}`;
+    const entry = inputList[idx];
+    const propPorts = getPropPort(idx, inputList);
+    const stepNode = {
+      nodeType: "step",
+      version: 1,
+      id: idNode,
+      name: stepTitle[entry.language],
+      note: {
+        version: 1,
+        id: idNote,
+        name: null,
+        description: null,
+        properties: {
+          UI_NOTE_PROP_HEIGHT: "0",
+          UI_NOTE_PROP_IS_EXPANDED: "false",
+          UI_NOTE_PROP_IS_STICKYNOTE: "false",
+          UI_NOTE_PROP_WIDTH: "0",
+        },
+      },
+      priority: idx,
+      properties: {
+        UI_PROP_COLORGRP: "0",
+        UI_PROP_IS_INPUT_EXPANDED: "false",
+        UI_PROP_IS_OUTPUT_EXPANDED: "false",
+        UI_PROP_NODE_DATA_ID: stepRef[entry.language],
+        UI_PROP_NODE_DATA_MODIFIED_DATE: nowTimestamp,
+        UI_PROP_XPOS: ((idx + 1) * NODE_SPACING).toString(),
+        UI_PROP_YPOS: "75",
+      },
+      portMappings: [
+        {
+          mappingType: "tableStructure",
+          portIndex: 0,
+          portName: "outTables",
+          tableStructure: {
+            columnDefinitions: null,
+          },
+        },
+      ],
+      stepReference: {
+        type: "uri",
+        path: `/dataFlows/steps/${stepRef[entry.language]}`,
+      },
+      arguments: {
+        codeOptions: {
+          code: entry.code,
+          contentType: "embedded",
+          logHTML: "",
+          resultsHTML: "",
+          variables: [
+            {
+              name: "_input1",
+              value: {
+                portIndex: 0,
+                portName: "inTables",
+                referenceType: "inputPort",
+              },
+            },
+            {
+              name: "_output1",
+              value: {
+                arguments: {},
+                portIndex: 0,
+                portName: "outTables",
+                referenceType: "outputPort",
+              },
+            },
+          ],
+        },
+      },
+    };
+    stepNode.properties = { ...stepNode.properties, ...propPorts };
+    flowData.nodes[idNode] = stepNode;
+
+    if (idx > 0) {
+      const connection = {
+        sourcePort: {
+          node: arrayIdNode[idx - 1],
+          portName: "outTables",
+          index: 0,
+        },
+        targetPort: {
+          node: arrayIdNode[idx],
+          portName: "inTables",
+          index: 0,
+        },
+      };
+      flowData.connections.push(connection);
+    }
+  }
+  return flowData;
+}
+
+export function convertSASNotebookToFlow(
+  content: string,
+  outputName: string,
+): Uint8Array {
+  const codeList = [];
+
+  try {
+    const notebookContent = JSON.parse(content);
+
+    for (const cell of notebookContent) {
+      let code = cell.value;
+      if (code !== "") {
+        const language = cell.language;
```

Comment:
so what about markdown? how about converting markdown to a Note?

Reply:
This is possible but needs to be further studied.
Notes are associated to a Node in SAS Studio but can be inserted at any stage in a Notebook.
Should we assume that all Markdown cells preceding (or following ?) a Code cell are related to that code and added as a Note to the Code Node ?
Markdown titles and subtitles (if available) could also be used as Swimlane / Node titles.

This feature would add its share of complexity should we implement it or keep it as a future improvement ?

Reply:
Yes I would go with the option of ignoring markdown for now, fixed in ef20e4fa61c12fab7ed885b70c890a7656c3668b. Thanks